sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2206,17 +2220,21 @@ class Analyzer(
        * @return (the [[Generator]], seq of output names, outer flag)
        */
       def unapply(e: Expression): Option[(Generator, Seq[String], Boolean)] = e match {
-        case Alias(GeneratorOuter(g: Generator), name) if g.resolved => Some((g, name :: Nil, true))
-        case MultiAlias(GeneratorOuter(g: Generator), names) if g.resolved => Some((g, names, true))
-        case Alias(g: Generator, name) if g.resolved => Some((g, name :: Nil, false))
-        case MultiAlias(g: Generator, names) if g.resolved => Some((g, names, false))
+        case Alias(GeneratorOuter(g: Generator), name) => Some((g, name :: Nil, true))
+        case MultiAlias(GeneratorOuter(g: Generator), names) => Some((g, names, true))
+        case Alias(g: Generator, name) => Some((g, name :: Nil, false))
+        case MultiAlias(g: Generator, names) => Some((g, names, false))
         case _ => None
       }
     }
 
     def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperatorsUp {
-      case Project(projectList, _) if projectList.exists(hasNestedGenerator) =>
-        val nestedGenerator = projectList.find(hasNestedGenerator).get
+      // We need to resolve all the functions that might be replaced with generators
+      // before validating a plan in this rule.
+      case p if p.expressions.exists(_.find(_.isInstanceOf[UnresolvedFunction]).isDefined) => p
+
+      case Project(projectList, _) if projectList.exists(hasUnsupportedNestedGenerator) =>
+        val nestedGenerator = projectList.find(hasUnsupportedNestedGenerator).get
         throw new AnalysisException("Generators are not supported when it's nested in " +
```

Comment:
Do we need to revise this message, `Generators are not supported when it's nested`?

Reply:
Sure.